unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Material/MaterialHandler.cs

```diff
@@ -34,7 +45,12 @@ public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
 
             while (activePromises.Count > 0)
             {
-                AssetPromiseKeeper_Material.i.Forget(activePromises.Dequeue());
+                var promise = activePromises.Dequeue();
+
+                if (isDebugMode)
+                    DataStore.i.sceneWorldObjects.RemoveMaterial(scene.sceneData.sceneNumber, entity.entityId, promise.asset.material);
```

Comment:
Same here,  `DataStore.i.sceneWorldObjects` can be injected

Reply:
Done!